plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java

```diff
@@ -124,18 +124,26 @@ protected IStatus doRun(IProgressMonitor monitor) {
 		return Status.OK_STATUS;
 	}
 
-	private boolean isUpdate(Connection connection, String project, String name) {
+	private boolean isUpdateThenDo(Connection connection, String project, String name) {
```

Comment:
is Update? then do update =) That's what i meant. It was isUpdate before and added update to the same method for not to get resources from connection 2 times.
I agree, that  `updateTriggerIfUpdate` is better